setup1-contributor/src/setup_keys/confirmation_key.rs

```diff
@@ -0,0 +1,84 @@
+use std::path::Path;
+
+use anyhow::Result;
+use blake2::{Blake2s, Digest};
+use serde::{Deserialize, Serialize};
+
+const CONFIRMATION_KEY_FILE: &str = ".confirmation_key";
+
+/// The data structure to persist the information
+/// about the private key in case of contributor restart.
+/// Private key is used to prove the participation
+/// in the setup
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct ConfirmationKey {
+    address: String,
+    private_key: String,
+}
+
+impl ConfirmationKey {
+    /// Return blake2s hash of confirmation key as hex string
+    ///
+    /// If there's `CONFIRMATION_KEY_FILE`, and the key_file.address
+    /// is the same as contributor address, use the file contents
+    /// to produce the hash.
+    ///
+    /// If there's no `CONFIRMATION_KEY_FILE` or key_file.address
+    /// is different, generate a new private_key, write it
+    /// to disk with contributor address, and use these private_key
+    /// and address to produce the hash
+    pub fn for_current_round(address: String) -> Result<String> {
+        let key_file = if Path::new(CONFIRMATION_KEY_FILE).exists() {
+            let key_file = Self::read_from_disk()?;
+            if key_file.address == address {
+                key_file
+            } else {
+                Self::create_new(address)?
```

Comment:
One less related question: would there be a situation that a user runs the entire system twice (under two different addresses), and the second time, the confirmation-key-file for the first time is overwritten? would it cause an issue? 

This does not need to be resolved in this PR.

Reply:
That's a great question! I paid a lot of attention to addressing this situation. There's a temporary file where I put address and confirmation key. If the user restarts the contributor, it will read the temporary file and compare the address. If the address is the same, then the old confirmation key is used. If the address is new (the user generated new keys), then the temporary file will be overwritten.

Reply:
Got it, so basically, if the user did not generate the new keys, the user can resume the contribution, under the same account as well as the confirmation key.

Reply:
Yes, that's the idea. And when the contribution is done, the temporary file is deleted, and the keys are printed out.